awx/main/dispatch/publish.py

```diff
@@ -83,6 +84,9 @@ def apply_async(cls, args=None, kwargs=None, queue=None, uuid=None, **kw):
                     'kwargs': kwargs,
                     'task': cls.name
                 }
+                guid = GuidMiddleware.get_guid()
+                if guid:
+                    obj['guid'] = guid
```

Comment:
When messages are *published* to the message bus, include the guid of the current thread (this is how the guid passes from e.g., an HTTP request or the task manager to the dispatcher)